### ISSUE TITLE
Exact position of invalid characters not reported by scanner

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [microsoft/TypeScript#43653]
# Bug Report

### 🔎 Search Terms

invalid character exact position scanner

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?ssl=1&ssc=1&pln=3&pc=1#code/Pcbo-QUAbghgTgAgB4G4HokA)

### 💻 Code

```ts
ï»¿
var x;   
```

### 🙁 Actual behavior

The errors are reported without any column information.

```
index.ts:1:2 - error TS1127: Invalid character.

1 ï»¿
   

index.ts:1:3 - error TS1127: Invalid character.

1 ï»¿
```
### 🙂 Expected behavior

TypeScript should show exactly which character is invalid, no?

```
index.ts:1:2 - error TS1127: Invalid character.

1 ï»¿
   ~


index.ts:1:3 - error TS1127: Invalid character.

1 ï»¿
    ~
```

Noticed this while investigating https://github.com/microsoft/TypeScript/pull/43005#discussion_r611858357...
